docs/resources/server/enchantments/builtin.md

```diff
@@ -0,0 +1,183 @@
+
+### Entity Effect Components
+[Entity Effect Components] are components that contain an `EnchantmentEntityEffect`, an extension of `EnchantmentLocationBasedEffect`. These override `EnchantmentLocationBasedEffect#onChangedBlock` to run `EnchantmentEntityEffect#apply` instead; this `apply` method is also directly invoked somewhere else in the codebase depending on the specific type of Entity Effect Component.
+
+Custom `EnchantmentEntityEffect` extensions can be registered through `BuiltInRegistries.ENCHANTMENT_ENTITY_EFFECT_TYPE`.
+
+Here is an example of the JSON definition of one such component from the Fire Aspect enchantment:
+```json5
+"minecraft:post_attack": [
+    {
+        "affected": "victim",
+        "effect": {
+            "type": "minecraft:ignite",
+            "duration": {
+                "type": "minecraft:linear",
+                "base": 4.0,
+                "per_level_above_first": 4.0
+            }
+        },
+        "enchanted": "attacker",
+        "requirements": {
+            "condition": "minecraft:damage_source_properties",
+            "predicate": {
+                "is_direct": true
+            }
+        }
+    }
+]
+```
+
+Here, the Entity Effect Component is `minecraft:post_attack`. Its effect is `minecraft:ignite`, which is implemented by the `Ignite` record. This record's implementation of `EnchantmentEntityEffect#apply` sets the target entity on fire.
+
+### Vanilla Enchantment Entity Effect Component Types
+#### Defined as `DataComponentType<List<ConditionalEffect<EnchantmentEntityEffect>>>`
+- `minecraft:hit_block`: Runs an entity effect when an entity (for example, a projectile) hits a block. Used by Channeling.
+- `minecraft:tick`: Runs an entity effect each tick. Used by Soul Speed.
+- `minecraft:projectile_spawned`: Runs an entity effect after a projectile entity has been spawned from a bow or crossbow. Used by Flame.
+
+#### Defined as `DataComponentType<List<TargetedConditionalEffect<EnchantmentEntityEffect>>>`
+- `minecraft:post_attack`: Runs an entity effect after an attack damages an entity. Used by Bane of Arthropods, Channeling, Fire Aspect, Thorns, and Wind Burst.
+
+For more detail on each of these, please look at the [relevant minecraft wiki page].
+
+### Attribute Effect Component
+The [Attribute Effect Component], `minecraft:attributes`, is a unique Enchantment Entity Effect Component of type `EnchantmentAttributeEffect` that is not registered as a Location Based Effect Component. It is used to apply attribute modifiers to the wielder of the enchantment, which are then removed when the enchanted item is no longer equipped. The JSON format is as follows:
+```json5
+"minecraft:attributes": [
+    {
+        "amount": {
+            "type": "minecraft:linear",
+            "base": 1,
+            "per_level_above_first": 1
+        },
+        "attribute": "namespace:class.attribute_name",
```

Comment:
```suggestion
        "attribute": "namespace:attribute_name",
```
The `class.` qualifier adds more confusion imo, Mojang keeping this qualifier is already confusing enough.